### PR TITLE
a script to check that edm::AssociationMap<>::transientMap_ is labebl…

### PR DESCRIPTION
**Pull request by gartung**
…ed transient=true in classes_def.xm

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

**Comment by Dr15Jones** [comment]
please test

File: DataFormats/TestObjects/src/classes_def.xml [hunk 1]
```diff
@@ -17,8 +17,9 @@
  <class name="edmtest::ArrayProduct" ClassVersion="10">
   <version ClassVersion="10" checksum="3635152897"/>
  </class>
- <class name="edmtest::TransientIntProduct" ClassVersion="10">
-  <version ClassVersion="10" checksum="304339852"/>
```
Comment (Dr15Jones) [inline DataFormats/TestObjects/src/classes_def.xml]: You must also keep this line

File: FWCore/Utilities/scripts/edmCheckClassTransients [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        try : p.Parse(nxml)
+        except xml.parsers.expat.ExpatError, e :
+            print '--->> edmCheckClassVersion: ERROR: parsing selection file ',self._file
+            print '--->> edmCheckClassVersion: ERROR: Error is:', e
+            raise
+        f.close()
+    def start_element(self,name,attrs):
+        if name in ('class','struct'):
+            if 'name' in attrs:
+                if re.search('^edm::AssociationMap<',attrs['name']):
```
Comment (Dr15Jones) [inline FWCore/Utilities/scripts/edmCheckClassTransients]: Let's make this more extensible. How about putting the pattern to search into a list and loop over the list? That way as new cases arise we just add to the list.

**Comment by Dr15Jones** [comment]
Could you please run this script on all DataFormats and SimDataFormats packages to see if we can find any problems?

File: FWCore/Utilities/scripts/edmCheckClassTransients (hunk 2 above)
Comment (gartung) [inline FWCore/Utilities/scripts/edmCheckClassTransients]: OK. I was thinking of making a dictionary of classes and transient fields.


On Sep 21, 2015, at 10:41 AM, Chris Jones <notifications@github.com<mailto:notifications@github.com>> wrote:


In FWCore/Utilities/scripts/edmCheckClassTransients<https://github.com/cms-sw/cmssw/pull/11397#discussion_r39986273>:

> +            if   (q1 or q2) and c == '<' : nxml += '&lt;'
> +            elif (q1 or q2) and c == '>' : nxml += '&gt;'
> +            # elif (q1 or q2) and c == '&' : nxml += '&amp;'
> +            else                         : nxml += c
> +            if c == '"' : q1 = not q1
> +            if c == "'" : q2 = not q2
> +        try : p.Parse(nxml)
> +        except xml.parsers.expat.ExpatError, e :
> +            print '--->> edmCheckClassVersion: ERROR: parsing selection file ',self._file
> +            print '--->> edmCheckClassVersion: ERROR: Error is:', e
> +            raise
> +        f.close()
> +    def start_element(self,name,attrs):
> +        if name in ('class','struct'):
> +            if 'name' in attrs:
> +                if re.search('^edm::AssociationMap<',attrs['name']):


Let's make this more extensible. How about putting the pattern to search into a list and loop over the list? That way as new cases arise we just add to the list.

Comment (Dr15Jones) [inline FWCore/Utilities/scripts/edmCheckClassTransients]: that sounds good

**Comment by gartung** [comment]
@Dr15Jones With the latest commits the script now runs on all packages under DataFormats and SimDataFormats and finds no errors.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
@gartung please put back the ```std::``` since in the future ROOT may require them. Instead, I suggest having your script strip them out.

**Comment by cvuosalo** [comment]
+1

For #11397 14091b9d98722328f047ca595b816e7cdddf2244

Updating classes_def.xml files in DataFormats. There should be no change in run-time behavior.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11397/8270/summary.html) against baseline CMSSW_7_6_X_2015-09-22-1100 show no significant differences, as expected.

**Comment by gartung** [comment]
Closing for the moment. If I use the generated rootmap file instead of the classes_def.xml file I should not need to make any changes to the existing classes_def.xml files.

**Pull request closed without merging**